feeds/wifi-ax/hostapd/patches/900-hapd-netlink-ubus-bridge.patch

```diff
@@ -0,0 +1,891 @@
+--- a/hostapd/main.c
++++ b/hostapd/main.c
+@@ -41,7 +41,7 @@ struct hapd_global {
+ static struct hapd_global global;
+ static int daemonize = 0;
+ static char *pid_file = NULL;
+-
++uint32_t cached_events_nr = 0;
+ 
+ #ifndef CONFIG_NO_HOSTAPD_LOGGER
+ static void hostapd_logger_cb(void *ctx, const u8 *addr, unsigned int module,
+--- a/src/ap/hostapd.h
++++ b/src/ap/hostapd.h
+@@ -97,6 +97,10 @@ struct hostapd_probereq_cb {
+ };
+ 
+ #define HOSTAPD_RATE_BASIC 0x00000001
++/* max number of cached events */
++#define HOSTAPD_MAX_CACHED_EVENTS 500
++/* event counter */
++uint32_t cached_events_nr;
+ 
+ struct hostapd_rate_data {
+ 	int rate; /* rate in 100 kbps */
+--- a/src/ap/ieee802_11.c
++++ b/src/ap/ieee802_11.c
+@@ -4928,6 +4928,7 @@ int ieee802_11_mgmt(struct hostapd_data
+ 	int ret = 0;
+ 	unsigned int freq;
+ 	int ssi_signal = fi ? fi->ssi_signal : 0;
++	int ubus_resp;
+ 
+ 	if (len < 24)
+ 		return 0;
+@@ -4997,30 +4998,61 @@ int ieee802_11_mgmt(struct hostapd_data
+ 	if (hapd->iconf->track_sta_max_num)
+ 		sta_track_add(hapd->iface, mgmt->sa, ssi_signal);
+ 
++	/* ubus request */
++	struct hostapd_ubus_request req = {
++		.type = HOSTAPD_UBUS_TYPE_MAX,
++		.mgmt_frame = mgmt,
++		.ssi_signal = ssi_signal
++	};
++
+ 	switch (stype) {
+ 	case WLAN_FC_STYPE_AUTH:
+ 		wpa_printf(MSG_DEBUG, "mgmt::auth");
+ 		handle_auth(hapd, mgmt, len, ssi_signal, 0);
++		hostapd_logger(hapd, NULL, HOSTAPD_MODULE_IEEE80211,
++			       HOSTAPD_LEVEL_DEBUG, "mgmt: [%d] [%s]",
++				   stype,
++			       "AUTH");
++		req.type = HOSTAPD_UBUS_AUTH_REQ;
+ 		ret = 1;
+ 		break;
+ 	case WLAN_FC_STYPE_ASSOC_REQ:
+ 		wpa_printf(MSG_DEBUG, "mgmt::assoc_req");
+ 		handle_assoc(hapd, mgmt, len, 0, ssi_signal);
++		hostapd_logger(hapd, NULL, HOSTAPD_MODULE_IEEE80211,
++			       HOSTAPD_LEVEL_DEBUG, "mgmt: [%d] [%s]",
++				   stype,
++			       "ASSOC");
++		req.type = HOSTAPD_UBUS_ASSOC_REQ;
+ 		ret = 1;
+ 		break;
+ 	case WLAN_FC_STYPE_REASSOC_REQ:
+ 		wpa_printf(MSG_DEBUG, "mgmt::reassoc_req");
+ 		handle_assoc(hapd, mgmt, len, 1, ssi_signal);
++		hostapd_logger(hapd, NULL, HOSTAPD_MODULE_IEEE80211,
++			       HOSTAPD_LEVEL_DEBUG, "mgmt: [%d] [%s]",
++				   stype,
++			       "RE-ASSOC");
++		req.type = HOSTAPD_UBUS_REASSOC_REQ;
+ 		ret = 1;
+ 		break;
+ 	case WLAN_FC_STYPE_DISASSOC:
+ 		wpa_printf(MSG_DEBUG, "mgmt::disassoc");
+ 		handle_disassoc(hapd, mgmt, len);
++		hostapd_logger(hapd, NULL, HOSTAPD_MODULE_IEEE80211,
++			       HOSTAPD_LEVEL_DEBUG, "mgmt: [%d] [%s]",
++				   stype,
++			       "DISASSOC");
++		req.type = HOSTAPD_UBUS_DISASSOC_REQ;
+ 		ret = 1;
+ 		break;
+ 	case WLAN_FC_STYPE_DEAUTH:
+ 		wpa_msg(hapd->msg_ctx, MSG_DEBUG, "mgmt::deauth");
+ 		handle_deauth(hapd, mgmt, len);
++		hostapd_logger(hapd, NULL, HOSTAPD_MODULE_IEEE80211,
++			       HOSTAPD_LEVEL_DEBUG, "mgmt: [%d] [%s]",
++				   stype,
++			       "DEAUTH");
```

Comment:
No req.type set here.  If not used, do we really need the log change?

Reply:
Removed.